### PR TITLE
First Kubelet metric and overall plumbing.

### PR DESCRIPTION
**Pull request by vmarmol**

### CODE REVIEW

**Comment by rjnagal** [comment]
curious: why are we pushing from kubelet to Prometheus when all other clients pull?

**Comment by vmarmol** [comment]
@rjnagal we're not pushing we're just exposing the HTTP endpoint. The monitoring framework will pull from the node.

File: pkg/kubelet/metrics/metrics.go [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		},
+	)
+	// TODO(vmarmol): Containers per pod
+	// TODO(vmarmol): Latency of pod startup
+	// TODO(vmarmol): Latency of SyncPods
+)
+
+func init() {
+	// Register the metrics.
+	prometheus.MustRegister(ImagePullLatency)
```
Comment (rjnagal) [inline pkg/kubelet/metrics/metrics.go]: Ah, client registration tripped me. Looks like this is just for adding a new collector type.

Comment (vmarmol) [inline pkg/kubelet/metrics/metrics.go]: Yep :)

**Comment by vishh** [comment]
LGTM. THis is super useful!

**Pull request merged**